pkg/executor/aggregate/agg_hash_executor.go

```diff
@@ -366,6 +399,15 @@ func (e *HashAggExec) fetchChildData(ctx context.Context, waitGroup *sync.WaitGr
 		if r := recover(); r != nil {
 			recoveryHashAgg(e.finalOutputCh, r)
 		}
+
+		// Wait for the finish of all partial workers
+		e.fetcherAndPartialSyncer.Wait()
+
+		if e.spillHelper.isSpillTriggered() && !e.spillHelper.checkError() {
+			// Spill the remaining data
+			e.spill()
+		}
+
 		for i := range e.partialInputChs {
 			close(e.partialInputChs[i])
 		}
```

Comment:
Can we put this for loop in line 402?

Reply:
> Can we put this for loop in line 402?

I think it's better not to put it in L402 as worker may still be running and we can ensure workers have exited in L411.